httpbp/middlewares.go

```diff
@@ -158,3 +165,43 @@ func InjectEdgeRequestContext(truster HeaderTrustHandler, impl *edgecontext.Impl
 		}
 	}
 }
+
+// SupportedMethods returns a middleware that checks if the request is made
+// using one of the given HTTP methods.
+//
+// Returns a raw, plain text 405 error response if the method is not supported.
+// If GET is supported, HEAD will be automatically supported as well.
+// Sets the "Allow" header automatically to the methods given.
+func SupportedMethods(method string, additional ...string) Middleware {
+	supported := make(map[string]bool, len(additional)+1)
+	supported[strings.ToUpper(method)] = true
+	for _, m := range additional {
+		supported[strings.ToUpper(m)] = true
+	}
+	if supported[http.MethodGet] {
+		supported[http.MethodHead] = true
+	}
+
+	allowed := make([]string, len(supported))
+	i := 0
+	for m := range supported {
+		allowed[i] = m
+		i++
+	}
+	sort.Strings(allowed)
+	allowedHeader := strings.Join(allowed, ",")
+
+	return func(name string, next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+			if ok := supported[r.Method]; !ok {
```

Comment:
nit: Here it can be just `if !supported[r.Method] {`

Reply:
ahh, meant to fix that 😬